src/message_type.rs

```diff
@@ -17,7 +17,164 @@
 
 //! Message types for PeerMessages
 
+#![allow(unknown_lints, derive_hash_xor_eq)]
+
 use std::fmt;
+use std::hash::{Hash, Hasher};
+
+use hex;
+use sawtooth_sdk::consensus::engine::PeerMessage;
+
+use crypto::digest::Digest;
+use crypto::sha2::Sha512;
+use error::PbftError;
+use protos::pbft_message::{PbftBlock, PbftMessage, PbftMessageInfo, PbftViewChange};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PbftMessageWrapper {
+    PbftMessage(PbftMessage),
+    PbftViewChange(PbftViewChange),
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct ParsedPeerMessage {
+    pub header_bytes: Vec<u8>,
+    pub header_signature: Vec<u8>,
+    pub message: PbftMessageWrapper,
+    pub message_bytes: Vec<u8>,
+}
+
+impl Hash for ParsedPeerMessage {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match &self.message {
+            PbftMessageWrapper::PbftMessage(m) => m.hash(state),
+            PbftMessageWrapper::PbftViewChange(m) => m.hash(state),
+        }
+    }
+}
+
+impl ParsedPeerMessage {
+    pub fn from_pbft_message(message: PbftMessage) -> Self {
+        Self {
+            message: PbftMessageWrapper::PbftMessage(message),
+            message_bytes: vec![],
+            header_bytes: vec![],
+            header_signature: vec![],
+        }
+    }
+
+    pub fn from_view_change_message(message: PbftViewChange) -> Self {
+        Self {
+            header_bytes: vec![],
+            header_signature: vec![],
+            message: PbftMessageWrapper::PbftViewChange(message),
+            message_bytes: vec![],
+        }
+    }
+
+    pub fn get_info(&self) -> &PbftMessageInfo {
```

Comment:
Rust convention is to not use the `get_` prefix for accessors (https://rust-lang-nursery.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter)

Reply:
I went with `get_info` to minimize the diff generated, since a fair number of places in the code already do something like `msg.get_info()`, and we're able to substitute the wrapper type without changing those. If people don't care about the larger diff I can switch that.

Reply:
I personally would prefer to adhere to Rust convention vs a smaller diff.